sycl/test/lit.cfg.py

```diff
@@ -85,6 +85,9 @@
 
 if triple == 'nvptx64-nvidia-cuda-sycldevice':
     config.available_features.add('cuda')
+    config.substitutions.append( ('%gpu_atomics_config', "-D__SYCL_EMULATE_FLOAT_ATOMICS__=1") )
```

Comment:
Shouldn't this macro be set somewhere in clang driver?

Reply:
There are a couple points here:
1. The macro itself is more of a temporary fallback that can be used until all non-"Intel GPU" targets come to support the new implementation provided by the headers.
2. The targets that would want this configuration are not limited by NVPTX - for example, support for CPU is also lacking. However, the user would usually compile for the "SPIR target" in both CPU and Intel GPU cases, so we wouldn't be able to detect their intention to use emulated atomic functions at the Clang driver level.

That said, I really like your idea in a sense that CLI sugar like `-fsycl-emulate-float-atomics` could be provided by the Clang driver, and would improve UX even for the temporary purposes. This would require an independent PR, though. While I'll be able to prepare such a PR quickly, what would be your take on having such an improvement reviewed & merged after this main changeset?

Reply:
Suggested PR: https://github.com/intel/llvm/pull/3030

Reply:
Thank you for the explanations, I though still have questions (do not consider them as stoppers for this PR).

What is 'nvptx64-nvidia-cuda-sycldevice' at the line 86 above? Isn't it some option or part of options to clang++ driver? If that is, why not to add that macro requiring emulation?
Ok, I guess that is doable, but then you say that CPU also does not support fast FP fetch_add, it only supports the emulation sequence. Wouldn't it be safer then to use the emulation sequence by default and true/fast FP fetch_add() by option like -fsycl-fast-float-atomics ?

Reply:
`nvptx64-*` is a valid target triple that is accepted by the Clang driver, exactly.

Basically, the primary target for the code containing such atomics are Intel GPUs, so these will be used in a majority of cases. Therefore, having this "new" behavior by default instead of the emulation sequence simplifies the usage for the main scenarios.

On the other hand, the goal with this new implementation of FP-typed `fetch_add`/`fetch_sub` is to have **all** of the targets eventually switched to accepting it. Therefore, adjustments for, say, NVPTX at the driver level would also be temporary - once https://github.com/intel/llvm/issues/2853 is resolved, we'd have to remove this internal driver setting. So a possible reason to avoid driver-level hard-coding is that it's longer to revert and release code changes than it is to stop using a particular macro/option on the client side. Again, with Intel GPUs as the primary target, this usage is likely to be modest.

Reply:
@v-klochkov, oh, now I'm finally getting your confusion as to this repo's lit configuration. The thing is, why do I even need to have an "emulated" compilation of the test if it's only launched on host?
I should simply remove this from here and only leave this configuration in the test-suite repo.

Reply:
Just for the record: my questions to this place can be formulated as:
1) Are we Ok that users' programs that used FP fetch_add() and passed previously on CPU (with emulation sequence) will start failing after re-compilation with new headers?
2) dealing with such funcs as fetch_add becomes difficult without having good documentation telling how to check if FP fetch_add() is supported on particular device or not.

Reply:
As per an offline discussion, the current conclusion (which remains up for debate) appears to be:
1. We are OK with that, as we're moving towards all platforms support of the SPIR-V extensions employed for FP atomics.
2. The quick solution to "unfathomable runtime failures" after recompilation for unsupported targets (primarily, CPU) is that we add a Clang preprocessor warning to suggest a solution. The guide on supporting/non-supporting targets will be added to the documentation/release notes.